src/Planners/src/QuinticSpline.cpp

```diff
@@ -0,0 +1,787 @@
+            this->addTripletCurrentKnot(absoluteKnotIndex, i * 2, 2 * (i - 1) + 2, tripletsList);
+        } else
+        {
+            this->addTripletPreviousKnot(absoluteKnotIndex, i * 2, 2 * (i - 1), tripletsList);
+            this->addTripletCurrentKnot(absoluteKnotIndex, i * 2, 2 * (i - 1) + 2, tripletsList);
+            this->addTripletNextKnot(absoluteKnotIndex, i * 2, 2 * (i - 1) + 2 + 2, tripletsList);
+        }
+    }
+
+    A.setFromTriplets(tripletsList.begin(), tripletsList.end());
+
+    // compute first coordinate
+    Eigen::VectorXd b(A.rows());
+    for (size_t j = 0; j < knots.front().position.size(); j++)
+    {
+        b.setZero();
+        for (size_t i = 0; i < numberOfInteriorKnots; i++)
+        {
+            const int absoluteKnotIndex = i + 1;
+
+            if (i == 0 && (i + 1) == numberOfInteriorKnots)
+            {
+                this->addKnownTermKnotPosition(absoluteKnotIndex, j, b.head<2>());
+                this->addKnownTermNextKnot(absoluteKnotIndex, j, b.head<2>());
+                this->addKnownTermPreviousKnot(absoluteKnotIndex, j, b.head<2>());
+
+            } else if (i == 0)
+            {
+                this->addKnownTermKnotPosition(absoluteKnotIndex, j, b.head<2>());
+                this->addKnownTermPreviousKnot(absoluteKnotIndex, j, b.head<2>());
+
+            } else if (i + 1 == numberOfInteriorKnots)
+            {
+                this->addKnownTermNextKnot(absoluteKnotIndex, j, b.segment<2>(i * 2));
+                this->addKnownTermKnotPosition(absoluteKnotIndex, j, b.segment<2>(i * 2));
+            }
+
+            else
+            {
+                this->addKnownTermKnotPosition(absoluteKnotIndex, j, b.segment<2>(i * 2));
+            }
+        }
+
+        Eigen::SparseQR<Eigen::SparseMatrix<double>,
+                        Eigen::COLAMDOrdering<Eigen::SparseMatrix<double>::StorageIndex>>
+            qrDecomposition;
+        qrDecomposition.compute(A);
+        Eigen::VectorXd solution = qrDecomposition.solve(b);
+
+        for(size_t i = 0; i < numberOfInteriorKnots; i++)
+        {
+            const int absoluteKnotIndex = i + 1;
+
+            knots[absoluteKnotIndex].velocity[j] = solution[i * 2];
+            knots[absoluteKnotIndex].acceleration[j] = solution[i * 2 + 1];
+        }
+    }
+}
+
+void QuinticSpline::Impl::getPositionAtTime(const double& t,
+                                            const Polynomial& poly,
+                                            Eigen::Ref<Eigen::VectorXd> position)
+{
+    // improve the readability of the code
+    const auto& a0 = poly.a0;
+    const auto& a1 = poly.a1;
+    const auto& a2 = poly.a2;
+    const auto& a3 = poly.a3;
+    const auto& a4 = poly.a4;
+    const auto& a5 = poly.a5;
+
+    position = a0
+             + a1 * t
+             + a2 * std::pow(t, 2)
+             + a3 * std::pow(t, 3)
+             + a4 * std::pow(t, 4)
+             + a5 * std::pow(t, 5);
```

Comment:
The time is going to be elevated at the power of 5. While walking we usually don't reset the time, so this number can grow very big and cause numerical issues. Consider defining a time offset according to the first knot point.

Reply:
Actually `getPositionAtTime` takes as input the current time minus the initial point instant. Let me know if I'm misunderstanding.


https://github.com/dic-iit/bipedal-locomotion-framework/blob/aaa044da923e6ed0dd895b5f293f36e24a5d39ab/src/Planners/src/QuinticSpline.cpp#L782-L784

Reply:
Ah great, I did not notice this was a method of ``Impl``. No issue then.